src/libos/src/process/do_spawn/mod.rs

```diff
@@ -233,7 +233,9 @@ fn new_process_common(
             };
             let user_stack_base = vm.get_stack_base();
             let user_stack_limit = vm.get_stack_limit();
-            let user_rsp = init_stack::do_init(user_stack_base, 4096, &argv, envp, &mut auxvec)?;
+            let init_stack_size = vm.get_stack_range().size() >> 8; // Use 1/256 of user stack range as init size (by default, 4M user stack -> 16K init size)
```

Comment:
I think using min(max(vm.get_stack_range().size() >> 8, 4096), vm.get_stack_range().size()) is more reasonable.

Reply:
Done. Thanks for your review.